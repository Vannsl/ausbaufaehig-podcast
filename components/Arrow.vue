<template>
  <i
    :class="direction"
    :style="{ marginTop: marginTop }"
    class="border-white group-hover:border-primary ml-2"
  />
</template>

<script>
export default {
  name: "Arrow",
  props: {
    direction: {
      type: String,
      required: true
    }
  },
  computed: {
    marginTop() {
      return this.direction === "up" ? "7px" : "";
    }
  }
};
</script>

<style>
i {
  border: solid;
  border-width: 0 1px 1px 0;
  display: inline-block;
  height: 5px;
  margin-top: 2px;
  padding: 3px;
  width: 5px;
}

.right {
  transform: rotate(-45deg);
}

.left {
  transform: rotate(135deg);
}

.up {
  transform: rotate(-135deg);
}

.down {
  transform: rotate(45deg);
}
</style>
