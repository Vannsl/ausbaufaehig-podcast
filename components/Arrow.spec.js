import { shallowMount } from "@vue/test-utils";
import Arrow from "./Arrow";

describe("Arrow", () => {
  const wrapper = shallowMount(Arrow, {
    propsData: {
      direction: 'up'
    },
  });

  it("renders the correct markup", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
