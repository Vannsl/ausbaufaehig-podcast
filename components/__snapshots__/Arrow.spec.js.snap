// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Arrow renders the correct markup 1`] = `
VueWrapper {
  "_emitted": Object {},
  "_emittedByOrder": Array [],
  "isFunctionalComponent": undefined,
}
`;
